tasks/polarisTask/util/PolarisPlatformSupport.ts

```diff
@@ -6,7 +6,22 @@ export default class PolarisPlatformSupport {
         if (platform == "win32") {
             return "/api/tools/" + client + "_cli-win64.zip";
         } else if (platform == "darwin") {
-            return "/api/tools/" + client + "_cli-macosx.zip";
+            // The mac runner architecture in the ADO returns same for both arm and non arm machines.
```

Comment:
minor: would've recommended to move this logic to a separate function, anyways would not block the MR for the same.